test/middleware.test.js

```diff
@@ -22,86 +22,79 @@ describe('loopback:middleware generator', function() {
-    var modelGen = givenMiddlewareGenerator();
-    helpers.mockPrompt(modelGen, {
-      name: 'my-middleware-2',
-      phase: '(custom phase)',
-      customPhase: 'my-phase-2',
-      nextPhase: 'routes',
-      paths: ['/x', '/y'],
-      params: '{"z": 1}',
-    });
-
-    modelGen.run(function() {
-      var newSources = Object.keys(readMiddlewaresJsonSync('server'));
-      var p1 = newSources.indexOf('my-phase-2');
-      var p2 = newSources.indexOf('routes');
-      expect(p1).to.equal(p2 - 1);
-      done();
-    });
+    return helpers.run(path.join(__dirname, '../middleware'))
+      .cd(SANDBOX)
+      .withPrompts({
+        name: 'my-middleware-1',
+        phase: 'my-phase',
+        paths: ['/x', '/y'],
+        params: '{"z": 1}',
+      }).then(function() {
+        var newSources = Object.keys(readMiddlewaresJsonSync('server'));
+        var expectedSources = builtinSources.concat(['my-phase']);
+        expect(newSources).to.have.members(expectedSources);
+      });
   });
 
-  it('adds a new entry to an existing phase in server/middleware.json',
-    function(done) {
-      var modelGen = givenMiddlewareGenerator();
-      helpers.mockPrompt(modelGen, {
-        name: 'my-middleware-3',
-        phase: 'routes',
+  it('adds a new phase next to a selected one', function() {
+    return helpers.run(path.join(__dirname, '../middleware'))
+      .cd(SANDBOX)
+      .withPrompts({
+        name: 'my-middleware-2',
+        phase: '(custom phase)',
+        customPhase: 'my-phase-2',
+        nextPhase: 'routes',
         paths: ['/x', '/y'],
         params: '{"z": 1}',
+      }).then(function() {
+        var newSources = Object.keys(readMiddlewaresJsonSync('server'));
+        var p1 = newSources.indexOf('my-phase-2');
+        var p2 = newSources.indexOf('routes');
+        expect(p1).to.equal(p2 - 1);
       });
+  });
 
+  it('adds a new entry to an existing phase in server/middleware.json',
+    function() {
       var builtinSources = Object.keys(
         readMiddlewaresJsonSync('server').routes
       );
-      modelGen.run(function() {
-        var newSources = Object.keys(readMiddlewaresJsonSync('server').routes);
-        var expectedSources = builtinSources.concat(['my-middleware-3']);
-        expect(newSources).to.have.members(expectedSources);
-        done();
-      });
-    });
-
-  it('supports sub-phase', function(done) {
-    var modelGen = givenMiddlewareGenerator();
-    helpers.mockPrompt(modelGen, {
-      name: 'my-middleware-4',
-      phase: 'routes',
-      subPhase: 'after',
-      paths: ['/x', '/y'],
-      params: '{"z": 1}',
+      return helpers.run(path.join(__dirname, '../middleware'))
+        .cd(SANDBOX)
+        .withPrompts({
+          name: 'my-middleware-3',
+          phase: 'routes',
+          paths: ['/x', '/y'],
+          params: '{"z": 1}',
+        }).then(function() {
+          // eslint-disable-next-line max-len
```

Comment:
Let's rewrite the next line so that it fits in our max line length.

```js
const middlewareConfig = readMiddlewaresJsonSync('server');
const newSources = Object.keys(middlewareConfig.routes);
```